z3c/dependencychecker/main.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+from z3c.dependencychecker.dependencychecker import logger
+from z3c.dependencychecker.dependencychecker import parse_command_line
+from z3c.dependencychecker.dependencychecker import set_log_level
+
+import os
+import pkg_resources
+import sys
+
+
+def main():
+    options, args = parse_command_line()
+    set_log_level(options.verbose)
+    path = get_path(args)
+    get_package_dependencies(path)
+
+
+def get_path(args):
+    """Get path from given arguments or use """
+    path = os.getcwd()
+    if len(args) > 0:
+        path = os.path.abspath(args[0])
+        if not os.path.isdir(path):
+            logger.info('Given path does not exist: {0}'.format(args[0]))
+            sys.exit(1)
+    logger.info('path used: %s', path)
+    return path
+
+
+def get_package_dependencies(path):
+    if not has_setup_py_file(path):
+        logger.info('No setup.py file found! on {0}'.format(path))
+        sys.exit(1)
+
+    new_path, filename = get_egg_info_folder(path)
```

Comment:
Egg info: do we need to support (or switch to) other approaches? wheels have another dir/file if I'm correct.

Reply:
So far we only allow to analyze source distributions that have been "enabled", i.e. ``python setup.py develop`` be it with buildout, pip or whatever.

Do we want to expand that to more approaches?

I was already planning to reformat the code, so we can have a generic entry point that looks at which kind of distribution are we getting and then call the right parser/extractor.